File: 322-coin-change/coin-change.cpp
class Solution {
public:
    int coinChange(vector<int>& coins, int amount) {
        vector<int> dp ( amount+1,amount+1);
        dp[0]=0;
        for(int i =1 ;i<=amount;i++)
        {
            for(int j =0 ; j< coins.size(); j++)
            {
                if(coins[j]<=i)
                {
                    dp[i] = min(dp[i],dp[i-coins[j]]+1);
                }
            }
        }
        if(dp[amount]!=amount+1) return dp[amount];
        else return -1;
    }
};

File: 322-coin-change/coin-change.java
class Solution {
    public int coinChange(int[] coins, int Amount) {
        int[] amount = new int[Amount+1];
        Arrays.fill(amount,Amount+1);
        amount[0]=0;
        for(int i = 1;i < Amount+1;i++)
        {
           for(int j = 0;j<coins.length;j++)
           {
            if(i>=coins[j])
            {
                amount[i] = Math.min(amount[i],amount[i-coins[j]]+1);
            }
           }
        }
        return (amount[Amount]==Amount+1)? -1:amount[Amount];
    }
}